model-optimizer/unit_tests/extensions/back/ChangeRandomUniformOutputType_test.py

```diff
@@ -0,0 +1,122 @@
+# Copyright (C) 2018-2021 Intel Corporation
+# SPDX-License-Identifier: Apache-2.0
+
+import unittest
+from argparse import Namespace
+
+import numpy as np
+
+from extensions.back.ChangeRandomUniformOutputType import ChangeRandomUniformOutputType
+from mo.graph.graph import Node
+from mo.utils.ir_engine.compare_graphs import compare_graphs
+from unit_tests.utils.graph import build_graph, result, connect, regular_op_with_shaped_data
+
+nodes = {
+    **regular_op_with_shaped_data('placeholder', [3], {'type': 'Parameter'}),
+    **regular_op_with_shaped_data('random_uniform', [3, 4, 5], {'type': 'RandomUniform', 'op': 'RandomUniform'}),
+    **regular_op_with_shaped_data('convert', [3, 4, 5], {'type': 'Convert'}),
+    **result('result'),
+
+    # new RandomUniform node and inputs
+    **regular_op_with_shaped_data('min_val', [1], {'type': 'Const'}),
+    **regular_op_with_shaped_data('max_val', [1], {'type': 'Const'}),
+    **regular_op_with_shaped_data('shape', [3], {'type': 'Const'}),
+}
+
+
+class ChangeRandomUniformOutputTypeTest(unittest.TestCase):
+    def test_fp32_to_fp16(self):
```

Comment:
We can easily collapse all the tests and use \@generator to merge all of the. Please, do it to decrease the file size